src/hooks/usePublisher.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useRef, useState } from "react";
+import { Director, Publish } from '@millicast/sdk';
+import type { Publisher, PublisherState } from './types/Publisher';
+
+
+const usePublisher = (token: string, streamName: string): Publisher => {
+
+    const [publisherState, setPublisherState] = useState<PublisherState>("ready");
+    const publisher = useRef<Publish>();
+
+    useEffect(() => {
+        const tokenGenerator = () => Director.getPublisher({ token: token, streamName: streamName });
+        publisher.current = new Publish(streamName, tokenGenerator);
```

Comment:
should return a cleanup function to stop the publisher if it's streaming

Reply:
Have a look please?